### PR TITLE
Add detail to tag descriptions

### PR DESCRIPTION
**Pull request by noon-dawg**
This pull request adds more detail to tag descriptions in `openapi/openapi.yaml.

This pull request also updates the capitalization of tag names. As a result, the related references in other files are also updated. This was completed in [9f767ed](https://github.com/Rebilly/api-definitions/pull/966/commits/9f767ed0f1480a4ad43d36a69e2be275e0d0384e) and [7aab864](https://github.com/Rebilly/api-definitions/pull/966/commits/7aab864a8e4b5d55747bd434d8ced460f7388787).

Also related to the change of tag name capitalization, there are 3 broken links in the docs repository. These are caused by references from the website repository that point to API tags that have been changed to lowercase. I corrected those links in another pull request, but this one needs to be merged first.

### CODE REVIEW

File: openapi/openapi.yaml [hunk 1]
```diff
@@ -49,381 +57,612 @@ components:
...
 
-      Transaction, Order, Plan, Product, Invoice, Note. That allows to
-      quickly find and use files related to
+      If `credit-file-proof` is requested in combination with
+      `identity-proof` and `address-proof`, `credit-file-proof`
+      is validated first, then the customer's identity and address
+      are validated. If all three forms of proof are valid, the
+      KYC request is fulfilled. If `identity-proof` or `address-proof`
+      documents do not pass validation, redirect your customer to the
+      KYC document gatherer to collect their KYC documents.
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:187]: That doesn't seem correct. 

In the case that a credit-file-proof is requested in combination with identity-proof and address-proof, the credit-file-proof is attempted first. If credit-file-proof is successful in validating the identity and address, then the KYC request is considered instantly fulfilled. If it is not, or if you the credit-file-proof isn't requested, then redirect your customer to the KYC document gatherer to collect their KYC documents.

File: openapi/openapi.yaml [hunk 2]
```diff
@@ -49,381 +57,612 @@ components:
...
+  - name: Customer authentication
+    description: >-
+      Validate the identity of users and manage authentication
+      credentials.
+  - name: Custom fields
+    description: |-
+      Use custom fields to extend a resource scheme to include custom
+      data that is not provided as a common field. Depending on the
+      resource on which the custom field is added, it will be
+      available in all related forms.
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:197]: It will be available in Rebilly dashboard, but not all related forms.

File: openapi/openapi.yaml [hunk 3]
```diff
@@ -49,381 +57,612 @@ components:
...
-      The Payment Instrument allows you to accept a variety of payment methods through a single API.
-  - name: Payment Tokens
-    description: >
-      Payment tokens are used to reduce the scope of PCI DSS compliance.  A
-      payment
+      Example: A custom field called `preferredCommunicationChannel`
+      is added to the customer resource. It has two allowed values,
+      which are 'email' and 'phone'. When new customers are added,
+      they will be prompted to provide this information.
+      Alternatively, if the customer is manually added, a field for
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:203]: >  When new customers are added, they will be prompted to provide this information.

That doesn't seem correct. Customers are not prompted for this information. It's up to developer on how he fills this data.

File: openapi/openapi.yaml [hunk 4]
```diff
@@ -49,381 +57,612 @@ components:
...
+        }
+      }
+      ```
 
-      authentication scheme in the Authentication section), which allows you to
+      For detailed information on Rebilly resources, see
+      [Resources](https://www.rebilly.com/docs/dev-docs/concepts/#resources).
+  - name: Disputes
+    description: |-
+      Use thes operations to manage disputes. A dispute occurs when a
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:218]:
```suggestion
      Use these operations to manage disputes. A dispute occurs when a
```

File: openapi/openapi.yaml [hunk 5]
```diff
@@ -49,381 +57,612 @@ components:
...
-      you do NOT want to do business with. They are a good tool for managing
-      risk. A blocklist entry that expires after a period of time we call a
-      greylist.
+    description: |-
+      Use blocklists help to prevent fraud and criminal activity.
+
+      Blocklists are lists of customer attribute values that are
+      blocked from buying from you. For example, if a customer
+      attempts to make a purchase from you with a credit card or
+      fingerprint that is in a blocklist, the transaction is blocked
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:80]: I am not sure how you can purchase with fingerprint. You can only block payment cards by adding payment card fingerprint to blocklist.

File: openapi/openapi.yaml [hunk 6]
```diff
@@ -49,381 +57,612 @@ components:
...
+      multipart/form-data encoded form, RAW POST &mdash; by sending
+      the file contents as the POST body, and fetching from URL
+      &mdash; by providing the file URL using the 'url' parameter.
 
-      sensitive cardholder info to your servers.  We recommend using this with
-      our
+      Use the attachments entity to link a file to one or multiple
+      objects, such as: customer, dispute, payment, transaction,
+      order, plan, product, invoice, or note. Attachments enable you
+      to quickly find and use files related to specific entities.
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:240]:
```suggestion
      Use the attachments entity to link a file to one or multiple
      objects, such as: customer, dispute, transaction,
      order, plan, product, invoice, or timeline comment. Attachments enable you
      to quickly find and use files related to specific entities.
```

File: openapi/openapi.yaml [hunk 7]
```diff
@@ -49,381 +57,612 @@ components:
...
+    description: |-
+      Use payment tokens to reduce the scope of
+      [PCI DSS compliance](https://www.rebilly.com/docs/pci-compliance/).
 
-      payment tokens.
+      A payment token can be made using a different authentication
+      scheme, which enables you to create a payment token directly
+      from the browser, and bypasses the need to send
+      sensitive cardholder info to your servers. We recommend using
+      this with the Rebilly.js library, which helps you set a form
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:276]:
```suggestion
      this with the FramePay library, which helps you set a form
```
?

File: openapi/openapi.yaml [hunk 8]
```diff
@@ -49,381 +57,612 @@ components:
...
-    description: |
-      Webhooks are designed to notify your systems when certain/all registered events happen in near real-time.
-      They allow you to collect information about those events. Rebilly can send these information via
-      POST to an URL of your choice.
+    description: |-
+      Use these operations to manage webhooks. Webhooks notify your
+      systems in real-time when certain events occur. For example,
+      when a new transaction occurs or a new subscription is created.
+      Webhooks enable you to collect information about events. Rebilly
+      sends this information by HTTP POST request to a defined URL of
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:476]:
```suggestion
      can send this information by HTTP POST request to a defined URL of
```

HTTP verb can be configured for a werbhook: GET, POST, PUT, PATCH, DELETE

File: openapi/openapi.yaml (hunk 1 above)
Comment (noon-dawg) [reply openapi/openapi.yaml]: Ah, I think I see where I went wrong here. I updated this in [2a1db2a](https://github.com/Rebilly/api-definitions/pull/966/commits/2a1db2a918553b0aec0eb1840803d65994a23c92)

File: openapi/openapi.yaml (hunk 3 above)
Comment (noon-dawg) [reply openapi/openapi.yaml]: I removed this line in [2a1db2a](https://github.com/Rebilly/api-definitions/pull/966/commits/2a1db2a918553b0aec0eb1840803d65994a23c92)

File: openapi/openapi.yaml (hunk 4 above)
Comment (noon-dawg) [reply openapi/openapi.yaml]: Added in [2a1db2a](https://github.com/Rebilly/api-definitions/pull/966/commits/2a1db2a918553b0aec0eb1840803d65994a23c92)

File: openapi/openapi.yaml (hunk 5 above)
Comment (noon-dawg) [reply openapi/openapi.yaml]: I removed this text

File: openapi/openapi.yaml (hunk 6 above)
Comment (noon-dawg) [reply openapi/openapi.yaml]: I added this change in [2a1db2a](https://github.com/Rebilly/api-definitions/pull/966/commits/2a1db2a918553b0aec0eb1840803d65994a23c92)****

File: openapi/openapi.yaml (hunk 7 above)
Comment (noon-dawg) [reply openapi/openapi.yaml]: Added in [2a1db2a](https://github.com/Rebilly/api-definitions/pull/966/commits/2a1db2a918553b0aec0eb1840803d65994a23c92)

File: openapi/openapi.yaml (hunk 8 above)
Comment (noon-dawg) [reply openapi/openapi.yaml]: Added in [2a1db2a](https://github.com/Rebilly/api-definitions/pull/966/commits/2a1db2a918553b0aec0eb1840803d65994a23c92)

File: openapi/openapi.yaml (hunk 2 above)
Comment (noon-dawg) [reply openapi/openapi.yaml]: Updated to read: "it will be available in the Rebilly UI."

File: openapi/openapi.yaml [hunk 9]
```diff
@@ -49,381 +57,629 @@ components:
...
+    description: |-
+      Use credit memos to provide a customer with store
+      credit. A common use case for using a credit memo is to provide
+      a customer with store credit, rather than a refund, if the
+      customer pays more than they owe or returns a product. For
+      information on the credit memo resource, see
+      [Resources](https://www.rebilly.com/docs/dev-docs/concepts/#resources).
+  - name: Payment instruments
+    description: |-
+      Use these operation to manage payment instruments. Payment
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:257]:
```suggestion
      Use these operations to manage payment instruments. Payment
```

File: openapi/openapi.yaml [hunk 10]
```diff
@@ -49,381 +57,629 @@ components:
...
+      [Reports API](https://reports-api-docs.rebilly.com).
   - name: Storefront API
-    description: |
-      The Rebilly Storefront API is currently experimental.  You may see
-      the [Storefront API Documentation here](https://storefront-api-docs.rebilly.com/).
-  - name: Users API
-    description: >
-      The Rebilly User API is primarily for our GUI apps.  You may see
+    description: |-
+      The Storefront API is a collection of transaction related
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:621]: Not sure if it is only transaction related.

Comment (ashkarpetin) [inline openapi/openapi.yaml:621]:
```suggestion
      The Storefront API is a collection of customer related
```

Not sure if it is only transaction related. It is a customer scoped API

File: openapi/openapi.yaml [hunk 11]
```diff
@@ -49,381 +57,629 @@ components:
...
-      determine whether someone is blocklisted or has political or economic sanctions against
-      them.
+    description: |-
+      Use Anti-Money Laundering (AML) operations to screen customers
+      and help prevent your business from becoming directly or
+      indirectly involved in criminal activity.
+
+      Use AML operations during customer creation, and some
+      transaction processing, to help determine if a potential
+      customer, or customer, has political or economic sanctions
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:67]:
```suggestion
      customer (lead), or customer, has political or economic sanctions
```

File: openapi/openapi.yaml [hunk 12]
```diff
@@ -49,381 +57,629 @@ components:
...
+      and adverse media lists.
   - name: Blocklists
-    description: |
-      Your blocklists contains values of customerIds, email addresses,
-      ipAddresses, bank identification numbers, countries or payment cards that
-      you do NOT want to do business with. They are a good tool for managing
-      risk. A blocklist entry that expires after a period of time we call a
-      greylist.
+    description: |-
+      Use blocklists help to prevent fraud and criminal activity.
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:75]:
```suggestion
      Use blocklists to prevent fraud and criminal activity.
```
Just suggestion

File: openapi/openapi.yaml [hunk 13]
```diff
@@ -49,381 +57,629 @@ components:
...
-        - purchase-proof
-        - funds-proof
-        - credit-file-proof
+    description: |-
+      Use these operations to manage customers. A customer is an
+      entity that purchases goods or services from you (a merchant),
+      and is the payee in any transaction that is credited to you.
+      Customers are associated with payment instruments,
+      subscriptions, invoices, and other miscellaneous relationship
+      models.
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:112]:
```suggestion
      subscriptions, invoices, and other related resources.
```

File: openapi/openapi.yaml (hunk 12 above)
Comment (noon-dawg) [reply openapi/openapi.yaml:75]: This sounds much stronger

File: openapi/openapi.yaml [hunk 14]
```diff
@@ -49,381 +57,632 @@ components:
...
+      goods or services.
   - name: Application owners
-    description: |
-      Creators of an organizations and organization owners in Rebilly.
+    description: |-
+      To use these operations, you must be an
+      application owner. Application owners are members of your company
+      or team that create and manage one or more
+      [organizations](https://www.rebilly.com/docs/settings/organizations-and-websites/)
+      in Rebilly.
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:672]: I think this part if incorrect, it doesn't have any relation with applications. Application owners API endpoints allow to register/retrieve application in the Rebilly App Store.

File: openapi/openapi.yaml [hunk 15]
```diff
@@ -49,381 +57,632 @@ components:
...
+      the applications that you own.
   - name: Application users
-    description: |
-      Users of an organization in Rebilly.
+    description: |-
+      To use these operations, you must be an
+      application user. Application users are members of your company
+      or team that use one or more
+      [organizations](https://www.rebilly.com/docs/settings/organizations-and-websites/)
+      in Rebilly.
```
Comment (ashkarpetin) [inline openapi/openapi.yaml:682]: I think this part if incorrect, it doesn't have any relation with applications. Application user API endpoints allow to retrieve/install/uninstall applications from the Rebilly App Store which were previosuly registered by `Apllication owners` endpoints.

https://www.rebilly.com/docs/dev-docs/install-an-app/

Comment (noon-dawg) [reply openapi/openapi.yaml]: Awesome input! Your comments helped me a lot 🙏

**Review by ashkarpetin — DISMISSED** [review]

**Review by Boydbueno — APPROVED** [review]

**Review by Boydbueno — APPROVED** [review]

**Review by ashkarpetin — APPROVED** [review]

File: openapi/openapi.yaml [hunk 16]
```diff
@@ -49,381 +57,628 @@ components:
...
+    description: |-
+      The Storefront API is a collection of customer related
+      endpoints and operations. This API is currently experimental.
 
-      the [User API Documentation
-      here](https://user-api-docs.rebilly.com).
+      Every action in the Rebilly
+      [frontend GUI application](https://app.rebilly.com/) is
+      supported by an API. These APIs are documented and you can use
+      them to automate any workflow.
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:627]:
```suggestion
```
I think this is not related to the tag described

File: openapi/openapi.yaml [hunk 17]
```diff
@@ -49,381 +57,628 @@ components:
...
+      The TaxJar integration enables you to calculate taxes and create
+      transactions for the Sales Tax AutoFile feature in your TaxJar
+      account. For more information, see
+      [TaxJar integration](https://www.rebilly.com/docs/automations/tax-jar-integration/#taxjar-integration).
   - name: Metadata
-    description: |
-      Metadata is data that provides structured information on other data.
+    description: >-
+      Use these operations to retrieve and manage metadata. Metadata
+      is data that structured information on other data.
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:564]:
```suggestion
      is data that structures information on other data.
```
not sure, but initial sentence sounds inconsistent

File: openapi/openapi.yaml [hunk 18]
```diff
@@ -49,381 +57,628 @@ components:
...
-      with a user's sign in credentials.
+      Use these operations to manage JWT sessions. A session contains
+      a token, which is a JSON Web Token. The token is created  with a
+      user's sign in credentials.
 
-      This token can be used to authenticate to the API.  In addition, the session can be set to
-      expire at a particular time. Use the token to then authenticate for further requests to the Rebilly API.
+      This token can be used to authenticate to the API. In addition,
+      the session can be set to expire at a particular time. Use the
+      token to authenticate for further requests to the Rebilly API.
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:446]:
```suggestion
      token to authenticate further requests to the Rebilly API.
```

File: openapi/openapi.yaml [hunk 19]
```diff
@@ -49,381 +57,628 @@ components:
...
-      Note: Organizations are shared between "Live" and "Sandbox" environments.
+      Note: Organizations are shared between the live and sandbox
+      environments.
   - name: Profile
-    description: |
-      A Profile represents the person currently logged in to Rebilly.
+    description: >-
+      Use these operations to manage user profiles. A profile
+      represents the person that is signed in to Rebilly.
   - name: Reset password
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:423]: Out of scope of the PR, but we seems to have similar tags:
`Password reset` and `Reset password`

File: openapi/openapi.yaml [hunk 20]
```diff
@@ -49,381 +57,628 @@ components:
...
-      Refund transactions.
+      Use these operations to set up payment instruments for payments,
+      authorize and hold funds, capture funds, make payments, make
+      payouts, and refund transactions.
   - name: Fees
-    description: |
-      Buy and sell fees.
+    description: |-
+      Use fees to reconcile transactions with applicable fees and
+      discount rates. Fees are not applied to directly to transaction
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:345]:
```suggestion
      discount rates. Fees are not applied directly to transaction
```

File: openapi/openapi.yaml [hunk 21]
```diff
@@ -49,381 +57,628 @@ components:
...
+      a digital payment, such as: credit cards, debit cards,
+      direct debits, payment service providers, and digital wallets.
 
-      Rebilly.js library, which helps you wire a form into this API resource and
-      create
+      A payment instrument is not a resource. It is a term used to
+      describe a specific set of resources, such as payment cards
+      and bank accounts. For more information on the payment
+      instruments, see
+      [Resources](https://www.rebilly.com/docs/dev-docs/concepts/#resources).
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:265]: This link seems incorrect

File: openapi/openapi.yaml [hunk 22]
```diff
@@ -49,381 +57,628 @@ components:
...
+  - name: Disputes
+    description: |-
+      Use these operations to manage disputes. A dispute occurs when a
+      customer contests a charge to their account. The dispute and
+      related information is made available to the merchant by the
+      bank or credit card company. The merchant then has the option to
+      represent the charge and win the case. This process is called
+      dispute resolution. If the merchant is unable to represent the
+      charge, the card issuer will typically rescind the sale and add
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:222]: `rescind` is a very specific word, can we replace it with more common?

File: openapi/openapi.yaml [hunk 23]
```diff
@@ -49,381 +57,628 @@ components:
...
+      available in the Rebilly UI.
 
-      token can be made using a different authentication scheme (refer to the
-      public key
+      Example: A custom field called `preferredCommunicationChannel`
+      is added to the customer resource. It has two allowed values,
+      which are 'email' and 'phone'.
 
-      authentication scheme in the Authentication section), which allows you to
+      ```yaml
```
Comment (VladimirBerdnik) [inline openapi/openapi.yaml:204]:
```suggestion
      ```js
```

File: openapi/openapi.yaml (hunk 16 above)
Comment (noon-dawg) [reply openapi/openapi.yaml:627]: Good point. This is marketing text and should not be in here

File: openapi/openapi.yaml (hunk 22 above)
Comment (noon-dawg) [reply openapi/openapi.yaml:222]: Excellent input! I updated this to:
> ...the card issuer will typically reverse the sale and add fees on top of the charge.
